examples/basics/export.py

```diff
@@ -0,0 +1,260 @@
+
+        k = tf.get_variable('filter', dtype=tf.float32,
+                            initializer=[[[[0.]], [[1.]], [[0.]]], [
+                                [[1.]], [[-4.]], [[1.]]], [[[0.]], [[1.]], [[0.]]]])
+        prediction_img = tf.nn.conv2d(img, k, strides=[1, 1, 1, 1], padding='SAME')
+        return prediction_img
+
+    def build_graph(self, input_img, target_img):
+
+        target_img = tf.cast(target_img, tf.float32)
+        target_img = tf.image.rgb_to_grayscale(target_img)
+
+        self.prediction_img = tf.identity(self.make_prediction(input_img), name='prediction_img')
+
+        cost = tf.losses.mean_squared_error(target_img, self.prediction_img,
+                                            reduction=tf.losses.Reduction.MEAN)
+        return tf.identity(cost, name='total_costs')
+
+    def optimizer(self):
+        lr = tf.get_variable('learning_rate', initializer=0.0, trainable=False)
+        return tf.train.AdamOptimizer(lr)
+
+
+def get_data(subset):
+    """
+    You might train this model using some fake data. But since the learning-rate is 0,
+    it does not change the model weights.
+    """
+    ds = FakeData([[SHAPE, SHAPE, CHANNELS], [SHAPE, SHAPE, CHANNELS]], 1000, random=False,
+                  dtype=['uint8', 'uint8'], domain=[(0, 255), (0, 10)])
+    ds = BatchData(ds, BATCH_SIZE)
+    return ds
+
+
+def get_config():
+    """
+    The usual config for training.
+    """
+    logger.auto_set_dir()
+
+    ds_train = get_data('train')
+
+    return TrainConfig(
+        model=Model(),
+        data=QueueInput(ds_train),
+        callbacks=[
+            ModelSaver(),
+        ],
+        steps_per_epoch=1,
+        max_epoch=1,
+    )
+
+
+class InferenceOnlyModel(Model):
+    """This illustrates the way to rewrite the inference graph to accept images encoded as base64.
+
+    Remarks:
+        CloudML expects base64 encoded data to feed into placeholders with suffix "_bytes".
+    """
+
+    def inputs(self):
+        # The inference graph only accepts a single image, which is different to the training model.
+        return [tf.placeholder(tf.string, (None,), 'input_img_bytes')]
+
+    def build_graph(self, input_img_bytes):
+        # prepare input (base64 encoded strings to images)
+        input_img = tf.map_fn(lambda x: tf.image.decode_png(x, channels=3), input_img_bytes, dtype=tf.uint8)
+
+        # It is a good idea to just copy the inference relevant parts to this graph.
+        prediction_img = self.make_prediction(input_img)
+
+        # outputs should be base64 encoded strings agains
+        prediction_img = tf.clip_by_value(prediction_img, 0, 255)
+        prediction_img = tf.cast(prediction_img, tf.uint8)
+        prediction_img_bytes = tf.map_fn(tf.image.encode_png, prediction_img, dtype=tf.string)
+
+        # prediction_img_bytes = tf.image.encode_png(prediction_img)
+        tf.identity(prediction_img_bytes, name='prediction_img_bytes')
+
+
+def export_npz():
```

Comment:
I would like to remove this function. Npz is used for the model zoo because it's easier to load and only one file. But I don't mean to make it feel like "standard for tensorpack" and users may use any format they like given the standard TF checkpoint support.
And this function isn't doing anything useful, after all.

Reply:
This was just a mental note for myself.